src/modules/folder/hooks/useNeedsToWait.jsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * When we mount the component when we already have data in cache,
+ * the mount is time consuming since we'll render at least 100 lines
+ * of File.
+ *
+ * React seems to batch together the fact that :
+ * - we change a route
+ * - we want to render 100 files
+ * resulting in a non smooth transition between views (Drive / Recent / ...)
+ *
+ * In order to bypass this batch, we use a state to first display a much
+ * more simpler component and then the files
+ */
+const useNeedsToWait = ({ isLoading }) => {
+  const [needsToWait, setNeedsToWait] = useState(true)
+  useEffect(() => {
+    let timeout = null
+    if (!isLoading) {
+      timeout = setTimeout(() => {
+        setNeedsToWait(false)
+      }, 50)
```

Comment:
Here are we waiting for 50ms in order to be in a new render loop? If yes we can reduce it to 1ms can't we?

Or maybe this is to display the skeleton long enough to avoid any flickering?

Reply:
I thinks this for avoid flickering but I only copy it from FolderViewBody trying to extract the essential piece. My commit message are enough explicite ☹️